airbyte-api/src/main/openapi/config.yaml

```diff
@@ -2177,6 +2457,65 @@ components:
           type: array
           items:
             $ref: "#/components/schemas/SourceDefinitionRead"
+    CustomSourceDefinitionCreate:
+      type: object
+      required:
+        - workspaceId
+        - sourceDefinition
```

Comment:
not sure if it's preferable to have the post body as specified here, which looks like
```
{
    "workspaceId": "<workspaceId>",
    "sourceDefinition": {
        "name": "<name>",
        "dockerRepository": "<dockerRepository>",
        "dockerImageTag": "<dockerImageTag>",
        // etc...
    }
}
```
or whether we should flatten it to
```
{
    "workspaceId": "<workspaceId>",
    "name": "<name>",
    "dockerRepository": "<dockerRepository>",
    "dockerImageTag": "<dockerImageTag>",
    // etc...
}
thoughts?
```

Reply:
I think I slightly prefer the first, because otherwise it implies that the `workspaceId` is a first-class property of the resource being created. The separation maps onto the underlying data model, a sourceDefinition that is created and then granted to a particular workspace

Reply:
cool, i agree!